internal/dag/gatewayapi_processor.go

```diff
@@ -69,20 +73,27 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 		// are associated with the Gateway. An empty Selector matches all routes.
 		for _, listener := range p.source.gateway.Spec.Listeners {
 
-			nsMatches, err := p.namespaceMatches(listener.Routes.Namespaces, route)
-			if err != nil {
-				p.Errorf("error validating namespaces against Listener.Routes.Namespaces: %s", err)
-			}
+			// Validate the Kind on the selector is a supported type.
+			switch listener.Routes.Kind {
```

Comment:
just looking at the docs (https://gateway-api.sigs.k8s.io/spec/#networking.x-k8s.io/v1alpha1.RouteBindingSelector), we should probably check `listener.Routes.Group` as well - sounds like empty string or `networking.x-k8s.io` would be valid here, anything else would not be valid.

Reply:
Yup sounds good! 👍